src/ast/mod.rs

```diff
@@ -960,9 +1145,10 @@ where
     Head: Clone + Display + ToSexp,
     Leaf: Clone + PartialEq + Eq + Display + Hash + ToSexp,
 {
+    // Not used
```

Comment:
why does this say not used? Shouldn't clippy catch that problem?

Reply:
I think it is never used now. The old code used this as a catch-all for non-relation functions. Now, the three subtypes are handled separately. However, it is still there to implement `Display`.

Reply:
If it's there to implement display then it's used
Perhaps change the comment to "used in implementation of `Display`"